pandas/core/nanops.py

```diff
@@ -829,9 +830,28 @@ def _checked_add_with_arr(arr, b):
 
     Raises
     ------
-    OverflowError if any x + y exceeds the maximum int64 value.
+    OverflowError if any x + y exceeds the maximum or minimum int64 value.
     """
-    if (np.iinfo(np.int64).max - b < arr).any():
-        raise OverflowError("Python int too large to "
-                            "convert to C long")
+    if np.__version__ >= LooseVersion('1.10.0'):
+        b2 = np.broadcast_to(b, arr.shape)
```

Comment:
i would put a comment back to this issue and s short expl of why this is being done (perf)

it's not obvious to someone looking at this code why (or even ehat) is happenening